### PR TITLE
feat: support reload scope's index.json if need

### PR DESCRIPTION
**Pull request by front-end-captain**
## Proposed Changes

-  support reload scope's index.json if current `scopeIndex` difference with <scope_folder>/index.json content

### CODE REVIEW

**Comment by GiladShoham** [comment]
Thanks
@davidfirst please have a look

**Comment by davidfirst** [comment]
@front-end-captain , the `index.json` file gets updated in-memory during objects writing and once done, it gets written the filesystem. I can't think of a scenario where the content in the filesystem is more recent than the content in-memory. 
If there is an issue with this file and you have steps to reproduce I'm happy to take a look. 
(also, in your PR I don't see where your new method gets called).

**Comment by front-end-captain** [comment]
@davidfirst Thank you for your reply. 

I am building my own remote scope server (not by way of docker), my service relies on @teambit/legacy to operate the remote bare scope, I have two instances of the service deployed in the cluster, but the same volume is mounted behind it, as shown here:
image

I found that when I exported new component to the service, occasionally the component were not fetched (via the graphql API), so I needed `scope` to expose a method to proactively reload an index.json while the service fetched the components via `scope.list` method

**Comment by davidfirst** [comment]
OK, I understand the scenario, thanks for the detailed explanation.
Unfortunately, this workflow of running multiple instances of the same scope is not supported. 
The issue you're now having with the index.json is probably just the start. For example, the scope instance has an internal cache, which gets cleared or replaced upon export and you're going to get undetermined results if the same components get accessed from another instance.

File: src/scope/objects/repository.ts [hunk 1]
```diff
@@ -82,6 +82,26 @@ export default class Repository {
...
+    if (force) {
+      this.scopeIndex = latestScopeIndex;
+      return;
+    }
+
+    const currentAllScopeIndexItems = this.scopeIndex.getAll();
+    const latestAllScopeIndexItems = latestScopeIndex.getAll();
+
+    if (currentAllScopeIndexItems.length !== latestAllScopeIndexItems.length) {
+      this.scopeIndex = latestScopeIndex;
```
Comment (davidfirst) [inline src/scope/objects/repository.ts:97]: you probably want to return here.

**Comment by front-end-captain** [comment]
@davidfirst Thanks for reviewing the code, the problem has been fixed, please check it again.

**Comment by front-end-captain** [comment]
@davidfirst If I encounter a problem with caching later, I will submit another PR. 😁

**Review by davidfirst — APPROVED** [review]

**Pull request merged**